modules/core/src/copy.cpp

```diff
@@ -563,25 +563,206 @@ Mat& Mat::setTo(InputArray _value, InputArray _mask)
     return *this;
 }
 
+#if CV_SIMD128
+template<typename V> CV_ALWAYS_INLINE void flipHoriz_single( const uchar* src, size_t sstep, uchar* dst, size_t dstep, Size size, size_t esz )
+{
+    typedef typename V::lane_type T;
+    int end = (int)(size.width*esz);
+    int width = (end + 1)/2;
+    int width_1 = width & -v_uint8x16::nlanes;
+    int i, j;
+
+    for( ; size.height--; src += sstep, dst += dstep )
+    {
+        for( i = 0, j = end; i < width_1; i += v_uint8x16::nlanes, j -= v_uint8x16::nlanes )
+        {
+            V t0, t1;
+
+            t0 = v_load((T*)((uchar*)src + i));
+            t1 = v_load((T*)((uchar*)src + j - v_uint8x16::nlanes));
```

Comment:
> v_uint8x16::nlanes

Looks like, there is hardcoded type in `template<typename V>` function.

Reply:
I am unable to reproduce it

Reply:
Can be reproduced with ENABLE_BUILD_HARDENING option ON:
```
cmake \
-GNinja   \
-DCMAKE_TOOLCHAIN_FILE=../opencv/platforms/linux/arm-gnueabi.toolchain.cmake   \
-DCMAKE_BUILD_TYPE=Debug   \
-DBUILD_LIST=ts,imgproc   \
-DCMAKE_INSTALL_PREFIX=install   \
-DENABLE_NEON=ON   \
-DCPU_BASLINE=NEON \
-DENABLE_BUILD_HARDENING=ON   \
../opencv
```

Reply:
Since ARM does NOT have an unaligned vector load would the best solution be?

```
diff --git a/modules/core/src/copy.cpp b/modules/core/src/copy.cpp
index 3f68a2555a..5e5f7ea025 100644
--- a/modules/core/src/copy.cpp
+++ b/modules/core/src/copy.cpp
@@ -574,7 +574,9 @@ template<typename V> CV_ALWAYS_INLINE void flipHoriz_single( const uchar* src, s

     for( ; size.height--; src += sstep, dst += dstep )
     {
-        for( i = 0, j = end; i < width_1; i += v_uint8x16::nlanes, j -= v_uint8x16::nlanes )
+        i = 0, j = end;
+#if !CV_NEON
+        for( ; i < width_1; i += v_uint8x16::nlanes, j -= v_uint8x16::nlanes )
         {
             V t0, t1;

@@ -585,6 +587,7 @@ template<typename V> CV_ALWAYS_INLINE void flipHoriz_single( const uchar* src, s
             v_store((T*)(dst + j - v_uint8x16::nlanes), t0);
             v_store((T*)(dst + i), t1);
         }
+#endif
         if (((size_t)src|(size_t)dst) % sizeof(T) == 0)
         {
             for ( ; i < width; i += sizeof(T), j -= sizeof(T) )
```